openssl-dynamic/src/main/c/ssl.c

```diff
@@ -439,9 +439,13 @@ static apr_status_t ssl_init_cleanup(void *data)
 #endif
     /* Corresponds to SSL_library_init: */
     EVP_cleanup();
-#if HAVE_ENGINE_LOAD_BUILTIN_ENGINES
+
+    // In case we loaded any engine we should also call cleanup. This is especialy important in openssl < 1.1.
+#ifndef OPENSSL_IS_BORINGSSL
+    // This is deprecated since openssl 1.1 but does not exist at all in BoringSSL.
     ENGINE_cleanup();
```

Comment:
Is this safe to call even if we didn't call ENGINE_load_builtin_engines()?

Reply:
Yes ... Just checked the source